### ISSUE TITLE
Reading incorrect values from PNG into float64 array

### ISSUE THREAD

**Issue by fab-jul** [imageio/imageio#302]
I'm trying to read the following file from the standard Kodak24 dataset (http://r0k.us/graphics/kodak/kodim03.html):

    im = imageio.imread('http://r0k.us/graphics/kodak/kodak/kodim03.png')

However, this results in float64 array containing incorrect values, instead of the expected uint8 array. I did not find a way to configure this. Any idea why this PNG fails? It also fails on all other images from the Kodak dataset.

Doing

    im = imageio.imread('imageio:chelsea.png')

works as expected, however, i.e., it produces a uint8 array.

**Comment by almarklein** [comment]
That is quite odd indeed. Thanks for reporting. I will have a look soonish. If you want to have a try at tracing this down, have a look at `plugins/pillow.py`. You could also try the Freeimage plugin to see what kind of data it produces.

**Comment by fab-jul** [comment]
Hi, thanks for the reply. Sadly I don't have the resources right now to debug this but here's some more info. I have to problems loading this image with

    from PIL import Image
    import scipy.misc

    im = scipy.misc.fromimage(Image.open(p))

So I'm thinking it is not directly related to PIL? The reason I switched from this is that `scipy.misc.fromimage` is getting deprecated

**Comment by almarklein** [comment]
Thanks for the update. Perhaps its a bug in the processing of the PIL image then.

**Comment by MareinK** [comment]
I also encountered this problem when working with [this image](https://upload.wikimedia.org/wikipedia/commons/thumb/8/8f/Stereogram_Tut_Random_Dot_Shark.png/400px-Stereogram_Tut_Random_Dot_Shark.png). I do get a warning when reading:

```
/usr/lib/python3.6/site-packages/imageio/core/util.py:104: UserWarning: Conversion from float64 to uint8, range [0.0, 255.0]
  'range [{2}, {3}]'.format(dtype_str, out_type.__name__, mi, ma))
```

**Comment by ugtony** [comment]
I use imageio.imread() to read an RGBA PNG image, and got a float64 array with wrong values as well. The loaded image is brighter than the original one.

**Comment by almarklein** [comment]
Reproduces, and found the issue. Will fix as part of #311.

The problem is occurring for images that have gamma in the metadata, which imageio uses to apply gamma correction (unless this is turned off via a kwarg).

**Comment by almarklein** [comment]
@MareinK I tweaked that warning a bit, and added a short notice that one should convert the image before saving it to suppress the warning.
